iib/web/api_v1.py

```diff
@@ -460,3 +460,103 @@ def regenerate_bundle_batch():
         ', '.join(request_id_strs),
     )
     return flask.jsonify(request_jsons), 201
+
+
+@api_v1.route('/builds/add-rm-batch', methods=['POST'])
+@login_required
+def add_rm_batch():
+    """
+    Submit a batch of requests to add or remove operators from an index image.
+
+    :rtype: flask.Response
+    :raise ValidationError: if required parameters are not supplied
+    """
+    payload = flask.request.get_json()
+    Batch.validate_batch_request_params(payload)
+
+    batch = Batch(annotations=payload.get('annotations'))
+    db.session.add(batch)
+
+    requests = []
+    # Iterate through all the build requests and verify that the requests are valid before
+    # committing them and scheduling the tasks
+    for build_request in payload['build_requests']:
+        if build_request.get('operators'):
+            # Check for the validity of a RM request
+            request = RequestRm.from_json(build_request, batch)
+        elif build_request.get('bundles'):
+            # Check for the validity of an Add request
+            request = RequestAdd.from_json(build_request, batch)
+        else:
+            raise ValidationError(
+                'Build request is not a valid Add/Rm request. This occurred on the build request'
+                f' in index {payload["build_requests"].index(build_request)}.'
+            )
+        db.session.add(request)
+        requests.append(request)
+
+    db.session.commit()
+    messaging.send_messages_for_new_batch_of_requests(requests)
+
+    request_jsons = []
+    # This list will be used for the log message below and avoids the need of having to iterate
+    # through the list of requests another time
+    request_id_strs = []
+    for build_request, request in zip(payload['build_requests'], requests):
+        request_jsons.append(request.to_json())
+        request_id_strs.append(str(request.id))
+
+        if isinstance(request, RequestAdd):
+            celery_queue = _get_user_queue()
```

Comment:
We could this once before entering the loop and use the value when calling `handle_add_request.apply_async` as well as `handle_rm_request.apply_async`.